### DISCUSSION TITLE
The battery revives itself after it's SOH==0?

### DISCUSSION THREAD

**Discussion by Hanyu-B** [pybamm-team/PyBaMM#2880] [Q&A]
I found it interesting that the battery revives during simulation. Here are the discharge capacity results plotted from the final solution of 3000 continuous cycles considering only SEI growth and Li plating (yes, I am trying to replicate the results in Yang 2017).

![image](https://user-images.githubusercontent.com/74037661/232404649-b5617a9b-1725-4a74-8131-146535b30da3.png)

As we can see from the plot, the discharge capacity went back to its original value after around 1400 cycles. And such processes repeat themselves in my 3000-cycle simulation. I am not sure what is really going on here, but the results look okay.

**Comment by valentinsulzer** [comment]
That looks very strange. It could be a bug. Can you share your code?

**Reply by Hanyu-B** [reply]
Thanks for the reply! Here is my code:

___________________________________________________________________________________________________________________________________________

paramss={'ec_diffusion':2e-18, 
        'kenetics_sei': 1e-16,
        'Lp_kinetics':1e-12,
        'ec_concentration':4565,
        'activation_energy': 2000,
} 

kenetics_sei=paramss["kenetics_sei"]
ec_diffusion=paramss["ec_diffusion"]
Lp_kinetics=paramss['Lp_kinetics']
ec_concentration=paramss['ec_concentration']
activation_energy=paramss['activation_energy']   

########### parameter setting ############
param = pybamm.ParameterValues("OKane2022")

##
############### Li-plating related
param.update({
    "Lithium plating transfer coefficient": 0.5,
    "Dead lithium decay constant [s-1]": 1E-6,

"Exchange-current density for plating [A.m-2]":0.00005
}, check_already_exists=False)

####################SEI related
param.update({"SEI kinetic rate constant [m.s-1]": kenetics_sei, 
'EC diffusivity [m2.s-1]':ec_diffusion,
'EC initial concentration in electrolyte [mol.m-3]': ec_concentration,
'SEI growth activation energy [J.mol-1]':activation_energy
}) 


########### Model setting ############
t_rest=5
v_min=2.8
v_max=4.2
pk1=[(

    
        f"Charge at 1C until {v_max} V",
        f"Hold at {v_max} V until 0.1 A",    
        f"Rest for {t_rest} minutes",
        f"Discharge at 2C until {v_min} V",  
        f"Rest for {t_rest} minutes"
        #
)] 

var_pts = {
"x_n": 20,  # negative electrode
"x_s": 20,  # separator
"x_p": 20,  # positive electrode
"r_n": 20,  # negative particle, n_Shells, this is especially needed if you are using state of charge dependent diffusion coefficients
"r_p": 20,  # positive particle, n_Shells
}

SEI_growth_option = ["reaction limited","ec reaction limited","solvent-diffusion limited","interstitial-diffusion limited","electron-migration limited","interstitial-diffusion limited", "ec reaction limited"]
plating_options = ["reversible", "irreversible", "partially reversible"]

model2 = pybamm.lithium_ion.DFN({
    "SEI": SEI_growth_option[1],
    "SEI film resistance":"distributed",
    "SEI porosity change":"true",
    "lithium plating":plating_options[1],
    "lithium plating porosity change":"true",
})

experiment = pybamm.Experiment(pk1*3000)#,termination="80% capacity")

sim_CCCV = pybamm.Simulation(model2, experiment=experiment, parameter_values=param,solver=pybamm.CasadiSolver(max_step_decrease_count=5), var_pts=var_pts)

sol_CCCV = sim_CCCV.solve(calc_esoh=True)

___________________________________________________________________________________________________________________________________________________

Actually, this problem would be masked if the termination condition was set.
